_posts/2015-04-03-april-newsletter.md

```diff
@@ -0,0 +1,476 @@
+
+This is a step towards
+@tarruda's goal to [remove all project specific types][long-types]. All that
+remains is `char_u` in which there are ~12,300 occurrences currently in use.
+
+### Adding a EndOfBuffer Highlight Group
+
+@mhinz wanted a way to [remove ~ after the file ends][highlight-idea]. He
+proposed a new [`:highlight`][highlight-docs] group, called
+[`EndOfBuffer`][highlight-endofbuffer], to go with the current [list of
+groups][highlight-groups].
+
+His [proposal][highlight-proposal] was first sent as a patch to
+[vim-dev][highlight-vim-dev] where his changes were accepted by Bram and should
+make it into Vim in the future. @mhinz let @fmoralesc prepare the [pull
+request][highlight-pr] to bring in the changes to Neovim. The pull request has
+since been merged into the master branch.
+
+### Garbage Collection Performance
+
+When using Vim with plugins that allocate a large number of objects there are
+times when the garbage collection will kick in and cause Vim to hang due to the
+time spent looking for objects to free.
+
+This issue was [posted][gc-original] to [vim\_dev][gc-vim-dev] by Yasuhiro
+Matsumoto. A [patch][gc-patch] as written by Ariya Mizutani and proposed to the
+mailing list which improves the algorithm for looking through the objects. In
+Yasuhiro's test case, it took unpatched Vim 100 seconds to execute but took just
+0.134 seconds when the patch was applied.
+
+An [issue][gc-issue] by @Shougo was created to discuss this patch being applied
+to Neovim. @oni-link then created a [pull request][gc-pr] to apply the patch
+which was merged shortly after.
+
+### Vimexpect and Example GDB Plugin
+
+@tarruda created a small Vimscript library in a [pull request][expect-pr] that
+mimics Tcl's [Expect][expect-wiki] for Neovim.
+
+The purpose of Expect (in Tcl) is to make it easy to script and automate text
+based applications that don't provide that functionality.
+
+@tarruda provided an example [plugin for GDB][expect-gdb] as a proof of concept.
+The plugin was created to make it easier to develop Neovim and show the
+vimexpect possibilities.
+
+### Options Updates
+
+#### Defaulting to `nocp`
+
+Vim has a [`cp`][options-cp-docs] option which lets Vim be more compatible with
+Vi. It can also be started by running `vim -C`.
+
+In December, @klusark made a [pull request][options-cp-pr] to change `nocp`,
+which is also known as no compatibility mode with Vi, to always be set.
+
+This means that `set nocp` and the flag [`-N`][options-cp-n] are non-operations.
+This also means that since there isn't a compatible mode, if you run `set cp` it
+will give an error.
+
+#### Removal of `edcompatible`
+
+An [issue][options-ed-issue] was created to discuss the removal of the
+[`edcompatible`][options-ed-docs] option. Turning on this feature gives special
+behavior for the `g` and `c` flags for the [`:substitute`][options-ed-s]
+command. The Vimdocs actually discourage the use of this option.
```

Comment:
Maybe `... command. Vim's documentation ...`? I know that Vimdocs is the name of the website, but it is after all just a frontend for Vim's documentation.